mpsl: cx: Add a generic coex variant using a single GPIO (#16716)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/wifi_coex.rst

.. table:: Generic one wire coexistence protocol pins

   ============  =========  =================================  ==============
   Pin           Direction  Description                        DT property
   ============  =========  =================================  ==============
   GRANT         In         Grant signal                       grant-gpios
   ============  =========  =================================  ==============

In cases where the GPIO is asserted after the radio activity has begun, the ``GRANT`` signal triggers a software interrupt, which in turn disables the radio.
No guarantee is made on the latency of this interrupt, but the ISR priority is configurable.

Enabling generic one wire coexistence
-------------------------------------

To enable the generic one wire coexistence, do the following:

1. Add the following node to the devicetree source file:

   .. code-block::

      / {
            nrf_radio_coex: radio_coex_one_wire {
               status = "okay";
               compatible = "generic-radio-coex-one-wire";
               grant-gpios =   <&gpio0 25 GPIO_ACTIVE_LOW>;
               concurrency-mode = <0>;
         };

>       };
> 
> #. Optionally, if not using the nRF91 Series SiP, the ``GRANT`` signal may be configured active-high using ``GPIO_ACTIVE_HIGH``
> #. Optionally, replace the node name ``radio_coex_one_wire`` with a custom one.
> #. The ``concurrency-mode`` property is optional and can be removed.
>    By default, or when set to 0, the ``GRANT`` signal will prevent both TX and RX.
>    When set to 1, it will only prevent TX.

Reviewer: ```suggestion
      };
      
   The ``concurrency-mode`` property is optional, and can be removed.
   By default, or when the property is set to ``0``, the ``GRANT`` signal will prevent both TX and RX.
   When set to ``1``, it will only prevent TX.

#. Optionally, if not using the nRF91 Series SiP, the ``GRANT`` signal may be configured active-high using ``GPIO_ACTIVE_HIGH``
#. Optionally, replace the node name ``radio_coex_one_wire`` with a custom one.
```
Moved the information from `concurrency-mode` to be a bit higher, since it's more information than a separate step.

> #. If not already present in the device tree, the GPIOTE interrupt may additionally be configured as follows (the first element is the IRQn, and the second is the priority):
> 
>    .. code-block::
> 
>       &gpiote {
>         interrupts = < 6 3 >;
>       };

Reviewer: ```suggestion
#. Optionally, if it is not already present in the device tree, you can configure the GPIOTE interrupt as follows:

   .. code-block::

      &gpiote {
        interrupts = < 6 3 >;
      };
      
   The first element is the IRQn, and the second one is the priority.
```
Is element the correct word here? Or should it be something like value?

Author: "element" is used below for grant-gpios so I just kept it
